jerry-core/jcontext/jcontext.h

```diff
@@ -116,6 +117,84 @@ typedef struct
 } jerry_context_t;
 
 /**
+ * Description of jerry instance, which contains the meta data and buffer.
+ */
+struct jerry_instance_t
+{
+  struct
+  {
+    uint32_t heap_size; /**< size of the heap */
+    uint8_t *context_p; /**< point to the entrance of the context in buffer */
+#ifndef JERRY_SYSTEM_ALLOCATOR
+    uint8_t *heap_p; /**< point to the entrance of the heap in buffer */
+#endif /* !JERRY_SYSTEM_ALLOCATOR */
+#ifndef CONFIG_ECMA_LCACHE_DISABLE
+    uint8_t *lcache_p; /**< point to the entrance of the lcache in buffer */
+#endif /* !CONFIG_ECMA_LCACHE_DISABLE */
+  } head;
+  uint8_t buffer[]; /**< the flexible array for storing context, heap and lcache */
```

Comment:
`buffer[]` is c99 syntax. https://en.wikipedia.org/wiki/Flexible_array_member

@martijnthe I will try your proposed `jerry_instance_t`. I think we can also remove the `head`, and let `heap_size`, `heap_p` ... be the first level member of jerry_instance